compiler/plugins/target/AMD-AIE/iree-amd-aie/Test/samples/matmul_pack_peel_objectfifo_e2e.mlir

```diff
@@ -32,3 +32,38 @@ func.func @matmul_i32(%lhs: tensor<128x256xi32>, %rhs: tensor<256x128xi32>) -> t
                     outs(%1: tensor<128x128xi32>) -> tensor<128x128xi32>
   return %res : tensor<128x128xi32>
 }
+
+// CHECK-LABEL: hal.executable.export public @matmul_truncf_bf16_bf16_dispatch_0_matmul_128x128x256_bf16xbf16xf32
+
+// CHECK:       aie.device(npu1_4col) {
+// CHECK-DAG:   %[[TILE_0_2:.+]] = aie.tile(0, 2)
+// CHECK-DAG:   %[[TILE_0_3:.+]] = aie.tile(0, 3)
+// CHECK-DAG:   %[[TILE_1_2:.+]] = aie.tile(1, 2)
+// CHECK-DAG:   %[[TILE_1_3:.+]] = aie.tile(1, 3)
+// CHECK-DAG:   %[[TILE_0_0:.+]] = aie.tile(0, 0)
+// CHECK-DAG:   %[[TILE_0_1:.+]] = aie.tile(0, 1)
+// CHECK-DAG:   aie.core(%[[TILE_0_2]])
+// CHECK-DAG:   aie.core(%[[TILE_1_2]])
+// CHECK-DAG:   aie.core(%[[TILE_0_3]])
+// CHECK-DAG:   aie.core(%[[TILE_1_3]])
+// CHECK-DAG:   aie.shim_dma_allocation {{.*}}(MM2S, 0, 0)
+// CHECK-DAG:   aie.shim_dma_allocation {{.*}}(MM2S, 1, 0)
+// CHECK-DAG:   aie.memtile_dma(%[[TILE_0_1]])
+// CHECK-DAG:   aie.mem(%[[TILE_0_2]])
+// CHECK-DAG:   aie.mem(%[[TILE_0_3]])
+// CHECK-DAG:   aie.mem(%[[TILE_1_2]])
+// CHECK-DAG:   aie.mem(%[[TILE_1_3]])
+// CHECK-DAG:   aie.shim_dma_allocation {{.*}}(S2MM, 0, 0)
+// CHECK:       {npu_instructions =
+// CHECK-SAME:   runtime_sequence_name = "matmul_truncf_bf16_bf16_dispatch_0_matmul_128x128x256_bf16xbf16xf32"
+func.func @matmul_truncf_bf16_bf16(%lhs: tensor<128x256xbf16>, %rhs: tensor<256x128xbf16>) -> tensor<128x128xbf16>
```

Comment:
Have you checked the numerics? It's okay to have one test in this directory, but I would move the matmul-elementwise test to a separate file.

For numerics check you can use this https://github.com/nod-ai/iree-amd-aie/blob/main/build_tools/ci/cpu_comparison/run.py by adding a new matmul-truncf file or template.